pkg/adapter/store.go

```diff
@@ -26,15 +25,13 @@ type Store struct {
 // NewStore creates a new Store with the ABCI prefix
 func NewStore(store ds.Batching) *Store {
 	return &Store{
-		Batching: kt.Wrap(store, &kt.PrefixTransform{
-			Prefix: ds.NewKey(KeyPrefix),
-		}),
+		Batching: store,
 	}
 }
 
 // LoadState loads the state from disk
 func (s *Store) LoadState(ctx context.Context) (*cmtstate.State, error) {
-	data, err := s.Get(ctx, ds.NewKey(stateKey))
+	data, err := s.Get(ctx, ds.NewKey(keyPrefix).Child(ds.NewKey(stateKey)))
```

Comment:
could we try to do this at the start, this way we avoid the extra allocations to create the prefix each time. 

in rollkit its done via 


func newPrefixKV(kvStore ds.Batching, prefix string) ds.Batching {
	return (ktds.Wrap(kvStore, ktds.PrefixTransform{Prefix: ds.NewKey(prefix)}).Children()[0]).(ds.Batching)
}

Reply:
The Wrap was used before and lead to the panic. Instead of NewKey.Child we can also just use a single `/abci/s` key. 

If your code snippet is used in rollkit, maybe the problem is there.  `.Children()[0]).(ds.Batching)` returns the original store without the prefix applied.